pkg/elfparser/elf.go

```diff
@@ -122,26 +122,35 @@ func newElfLoader(elfFile *elf.File, bpfmapapi ebpf_maps.BpfMapAPIs, bpfprogapi
 	return elfloader
 }
 
-func (b *bpfSDKClient) LoadBpfFile(path, customizedPinPath string) (map[string]BpfData, map[string]ebpf_maps.BpfMap, error) {
-	bpfFile, err := os.Open(path)
+// LoadBpfFromReader loads ELF file from io.ReaderAt and parses an ELF file into BPF objects
+func (b *bpfSDKClient) LoadBpfFromReader(r io.ReaderAt, customizedPinPath string) (
```

Comment:
You need to add it to the - interface `type BpfSDKClient `

Reply:
Oh! Thank you for pointing that out. 
Indeed, we needed to add the interface. I've fixed to add.